app/controllers/documents_controller.rb

```diff
@@ -10,6 +10,10 @@ class DocumentsController < ApplicationController
   before_action :set_navbar_actions,    except: [:edit, :plaintext]
   before_action :set_footer_visibility, only:   [:edit]
 
+  # Skip UI-heavy calls for API endpoints
+  skip_before_action :cache_most_used_page_information, only: [:update]
```

Comment:
L14-15 are the only real changes we need out of this PR. We can merge or cherry-pick and then close this.